back/src/config/pg/migrations/1723126978255_remove-counsellor-rights-on-users.ts

```diff
@@ -0,0 +1,22 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import { MigrationBuilder } from "node-pg-migrate";
+
+export async function up(pgm: MigrationBuilder): Promise<void> {
+  await pgm.sql(`
+    WITH agenciesWithTwoStepsValidation as (
+        SELECT agency_id
+        FROM "users__agencies"
+        WHERE roles::text like '%counsellor%'
+          AND is_notified_by_email is true
+    ), usersToUpdate as (SELECT user_id, agency_id
+    FROM "users__agencies"
+    where roles::text like '%counsellor%'
+      AND agency_id not in (select * from agenciesWithTwoStepsValidation))
+    UPDATE users__agencies
+    SET roles = '["agency-viewer"]'::jsonb
+    FROM usersToUpdate
+    WHERE users__agencies.user_id = usersToUpdate.user_id and users__agencies.agency_id = usersToUpdate.agency_id
```

Comment:
```suggestion
    WITH 
    agenciesWithTwoStepsValidation as (
      SELECT agency_id
      FROM "users__agencies"
      WHERE roles::text LIKE '%counsellor%'
      AND is_notified_by_email IS true
    ), 
    usersToUpdate as (
      SELECT user_id, agency_id
      FROM "users__agencies"
      WHERE roles::text LIKE '%counsellor%'
      AND agency_id NOT IN (
        SELECT * 
        FROM agenciesWithTwoStepsValidation
      )
    )
    UPDATE users__agencies
    SET roles = '["agency-viewer"]'::jsonb
    FROM usersToUpdate
    WHERE users__agencies.user_id = usersToUpdate.user_id 
    AND users__agencies.agency_id = usersToUpdate.agency_id
```
C'est juste de la mise en forme pour mieux comprendre l'intention.
Peut être renommer usersToUpdate en userRightsOnAgenciesWithOneStepValidation

Mais attention avec cette query, il faut bien s'assurer de son comportement sinon on défonce les droits de plein de monde.